docs/migration/4_0.md

```diff
@@ -1,3 +1,35 @@
 # Technical Changes in Spartacus 4.0
 
 ## Breaking Changes Introduced in 4.0
+
+## Automated Migrations for Version 4.0
+
+- `ProductVariantsModule` was removed from @spartacus/storefront. Use `@spartacus/product/variants` feature-library instead.
+- `ProductVariantsComponent` was removed from @spartacus/storefront. Use `ProductVariantsContainerComponent` from `@spartacus/product/variants/components` as a replacement.
+- `ProductVariantsModule` was removed from @spartacus/storefront. Use `ProductVariantsContainerModule` from `@spartacus/product/variants/components` as a replacement.
+  
+- `VariantColorSelectorComponent` was removed from @spartacus/storefront. Use `ProductVariantColorSelectorComponent` from `@spartacus/product/variants/components` as a replacement.
+- `VariantColorSelectorModule` was removed from @spartacus/storefront. Use `ProductVariantColorSelectorModule` from `@spartacus/product/variants/components` as a replacement.
+  
+- `VariantSizeSelectorComponent` was removed from @spartacus/storefront. Use `ProductVariantSizeSelectorComponent` from `@spartacus/product/variants/components` as a replacement.
+- `VariantSizeSelectorModule` was removed from @spartacus/storefront. Use `ProductVariantSizeSelectorModule` from `@spartacus/product/variants/components` as a replacement.
+  
+- `VariantStyleSelectorComponent` was removed from @spartacus/storefront. Use `ProductVariantStyleSelectorComponent` from `@spartacus/product/variants/components` as a replacement.
+- `VariantStyleSelectorModule` was removed from @spartacus/storefront. Use `ProductVariantStyleSelectorModule` from `@spartacus/product/variants/components` as a replacement.
+  
+- `VariantStyleIconsComponent` was removed from @spartacus/storefront. Use `ProductVariantStyleIconsComponent` from `@spartacus/product/variants/root` as a replacement.
+- `ProductVariantStyleIconsComponent` was moved from `@spartacus/product/variants/components`to `@spartacus/product/variants/root` instead.
```

Comment:
Can you add also add migration for this case?

Reply:
Done